docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,518 @@
+your data, but can impact search performance based on the computation defined in
+the runtime script.
+
+To balance search performance and flexibility, index fields that you'll
+commonly search for and filter on, such as a timestamp. {es} automatically uses
+these indexed fields first when running a query, resulting in a fast response
+time. You can then use runtime fields to limit the number of fields that {es}
+needs to calculate values for. Using indexed fields in tandem with runtime
+fields provides flexibility in the data that you index and how you define
+queries for other fields.
+
+Use the <<async-search,asynchronous search API>> to run searches that include
+runtime fields. This method of search helps to offset the performance impacts
+of computing values for runtime fields in each document containing that field.
+
+IMPORTANT: Queries against runtime fields are considered expensive. If
+<<query-dsl-allow-expensive-queries,`search.allow_expensive_queries`>> is set
+to `false`, expensive queries are not allowed and {es} will reject any queries
+against runtime fields.
+
+[discrete]
+[[runtime-mapping-fields]]
+=== Mapping a runtime field
+You map runtime fields by adding a `"runtime"` section under the mapping
+definition. Within that section, you
+<<modules-scripting-using,define a Painless script>>, which has access to the
+original `_source` and `doc_values` (the mapped field plus its values). At
+search time, the script runs and generates values for each scripted field.
+
+NOTE: You can define a runtime field in the mapping definition without a
+script. {es} will look in `_source` for a field with the same name as the
+runtime field and use values from that field at query time.
+
+Runtime fields are similar to the <<script-fields,`script_fields`>> parameter
+of the `_search` request, but also make the script results available for
+queries and aggregations.
+
+The script in the following request extracts the day of the week from the
+`@timestamp` field, which is defined as a `date` type:
+
+[source,console]
+----
+PUT /my-index
+{
+  "mappings": {
+    "runtime": { <1>
+      "day_of_week": {
+        "type": "keyword", <2>
+        "script": { <3>
+          "source": "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+        }
+      }
+    },
+    "properties": {
+      "timestamp": {"type": "date"}
+    }
+  }
+}
+----
+
+<1> Runtime fields are defined in the `"runtime"` section of the mapping
+definition.
+<2> Each runtime has its own field type, just like any other field.
+<3> The script defines the evaluation to calculate at search time.
+
+The `"runtime"` section supports `boolean`, `date`, `double`, `ip`, `keyword`,
+and `long` data types. Runtime fields with a `type` of `date` can accept the
+<<mapping-date-format,`format`>> parameter exactly as the `date` field type.
+
+IMPORTANT: Objects are not supported in the `"runtime"` section under the
```

Comment:
this makes it sound like a limitation, while it is not. I also don't understand the second part: you don't need to shadow, nor define the field in the search request if it is part of an object. I think that you are rather trying to explain that when you want to map a runtime field that is part of some object structure e.g. object.field , you don't need to recreate the object structure in the runtime field, but rather just define the leaf runtime field as object.field. Correct?

Reply:
I added a new section in [Mapping runtime fields](https://elasticsearch_62653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime-mapping-fields.html) to include using dot notation when mapping runtime fields within objects. I included a heading to separate this information within the broader page for mapping a runtime field, and that heading will be included in the next commit:
![image](https://user-images.githubusercontent.com/25848033/101647917-49aa1f00-3a07-11eb-949a-5afe0c5d4a85.png)

<img src=https://user-images.githubusercontent.com/25848033/101647917-49aa1f00-3a07-11eb-949a-5afe0c5d4a85.png>